### PR TITLE
add 374. Guess Number Higher or Lower

### PR DESCRIPTION
**Pull request by tkkawa**
Please check my code.

### CODE REVIEW

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
```py
class Solution:
    def guessNumber(self, n: int) -> int:
        left = 1
        right = n + 1
        
        while left < right:
            mid = left + right >> 1
            comp = guess(mid)
            
            if comp == 0:
                return mid
            
            if 0 < comp:
                left = mid + 1
            else:
                right = mid
                
        return right - 1
```

**Comment by tkkawa** [comment]
I modified the code.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
modify this

**Comment by tkkawa** [comment]
I modified the code.
Please check my code.

**Review by ryuji0123 — APPROVED** [review]

File: 374_guess-number-higher-or-lower.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+            if comp == 0:
+                return mid
+
+            if 0 < comp:
+                left = mid
+            else:
+                right = mid
+
+        return right - 1
```
Comment (ryuji0123) [inline 374_guess-number-higher-or-lower.py:28]: return left

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by tkkawa** [comment]
I modified the code.
Please check my code.

**Review by kobakobashu — APPROVED** [review]
LGTM

**Pull request closed without merging**